controller/controlplane_extensions/metricsscraper/manager.go

```diff
@@ -291,9 +291,9 @@ func (msm *Manager) NotifyRemove(ctx context.Context, cp types.NamespacedName) {
 // that is associated with the given ControlPlane.
 // It returns true if the scraper was added.
 func (msm *Manager) Add(cp *gwtypes.ControlPlane, pipeline MetricsScrapePipeline) bool {
-	// TODO(pmalek): implement DataPlane external URL type
-	// ref: https://github.com/kong/kong-operator/issues/1366
-	if cp == nil || cp.Spec.DataPlane.Type != gwtypes.ControlPlaneDataPlaneTargetRefType || cp.Spec.DataPlane.Ref == nil {
+	if cp == nil ||
+		((cp.Spec.DataPlane.Type != gwtypes.ControlPlaneDataPlaneTargetRefType || cp.Spec.DataPlane.Ref == nil) &&
```

Comment:
The compound of `||` s and `&&`s looks complex here (3 layers: `() || ( ( || ) && () ) `). Should we add comments or use separate `if`s here?

Reply:
WDYT about using a `switch` d7c3e2283f234728886c1694ebb677792abec5d7 ?